client/ide/lib/realtimemanagerfactory.coffee

```diff
@@ -8,8 +8,363 @@ module.exports = class RealtimeManagerFactory extends KDObject
 
   get: (type) ->
       if type is 'FIREBASE'
-        realtime = new FirebaseRealtimeManager()
+        realtime = new FirebaseRealtimeManager
       else if type is 'GOOGLE_DRIVE'
-        realtime = new GoogleDriveRealtimeManager()
+        realtime = new GoogleDriveRealtimeManager
       return realtime
   
+  setRealtimeDoc: (realtimeDoc) ->
+
+    @realtimeDoc = realtimeDoc
+    
+  getRealtimeDoc: ->
+
+    unless @realtimeDoc
+      throw new Error 'RealtimeDoc is not set yet for FirebaseRealtimeManager'
+
+    return @realtimeDoc
+    
+  createFile: (options, callback) ->
+
+    { title, preventEvent } = options
+
+    return throw new Error 'title is required'  unless title
+
+    options      =
+      resource   :
+        mimeType : 'application/vnd.google-apps.drive-sdk'
+        title    : title
+    if type is 'FIREBASE'
+      firebase.put(options).execute (file) =>
+      callback null, file
+    else if type is 'GOOGLE_DRIVE'
```

Comment:
these shouldn't be handled like this. you need to implement an interface which will be base for each realtime provider and after that you need to implement them in place in their classes;

Here is an example;

### Interface
```coffee
class RealTimeInterface
  
  createFile: (options, callback) ->
    callback { message: 'createFile: Not implemented' }

  getFile: (options, callback) ->
    callback { message: 'getFile: Not implemented' }
...
```

### GoogleDrive Implementation
```coffee
class GoogleDriveInterface extends RealTimeInterface
  
  createFile: (options, callback) ->
    options      =
      resource   :
        mimeType : 'application/vnd.google-apps.drive-sdk'
        title    : title
    gapi.client.drive.files.insert(options).execute (file) ->
      callback null, file

  getFile: (options, callback) ->
    fileId = options.id
    gapi.client.drive.files.get({ fileId }).execute (file) =>
      @emit 'FileFetched', file
      @loadFile { id: file.id }, callback
...
```

### Firebase Implementation
```coffee
class FirebaseInterface extends RealTimeInterface
  
  createFile: (options, callback) ->
    options      =
      resource   :
        mimeType : 'application/vnd.google-apps.drive-sdk'
        title    : title 
    firebase.put(options).execute (file) ->
      callback null, file

  getFile: (options, callback) ->
    fileId = options.id
    firebase.get({ fileId }).execute (file) => 
      @emit 'FileFetched', file
      @loadFile { id: file.id }, callback...
...
```

you need to change all other methods has `if` blocks in it for provider based implementations.